### PR TITLE
Setup lint staged

### PR DESCRIPTION
**Pull request by amit-mb**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +35,9 @@
     "enzyme-adapter-react-16": "^1.1.1",
     "enzyme-adapter-react-helper": "^1.2.3",
     "fork-ts-checker-webpack-plugin": "^0.4.1",
+    "husky": "^0.14.3",
     "jest": "^22.1.4",
+    "lint-staged": "^7.2.0",
```
Comment (revathskumar) [inline package.json]: Shouldn't we add `prettier` as `devDependencies`.

File: package.json [hunk 2]
```diff
@@ -66,8 +72,21 @@
     "build": "webpack -p --config webpack.config.prod.js",
     "watch": "webpack --config webpack.config.dev.js --watch --progress --colors",
     "test": "jest",
-    "test:watch": "jest --watch"
+    "test:watch": "jest --watch",
+    "precommit": "lint-staged"
   },
+ "lint-staged": {
+   "linters":{
+       "*.{js,jsx,css}": [
```
Comment (revathskumar) [inline package.json]: Does prettier format css as well?

Comment (revathskumar) [reply package.json]: Do we have any `jsx` component?

File: package.json [hunk 3]
```diff
@@ -66,8 +72,21 @@
...
+        "prettier --config .prettierrc --write",
+        "git add"
+      ],
+       "*.{ts,tsx}": ["prettier --config .prettierrc --parser typescript --write", "git add", "jest"]
+   },
+   "ignore": [
+    "/node_modules/"
+  ]
```
Comment (revathskumar) [inline package.json]: fix indendation

File: package.json [hunk 4]
```diff
@@ -66,8 +72,21 @@
...
+},
```
Comment (revathskumar) [inline package.json]: Fix indendation

File: package.json (hunk 2 above)
Comment (amit-mb) [reply package.json]: checked prettier, it supports `css` formatting with a plugin. removing it for now. Maybe we can use [stylelint](https://stylelint.io/) or [postcss](https://postcss.org/)

File: package.json (hunk 1 above)
Comment (amit-mb) [reply package.json]: Do we need it? `prettier` is a devDependency for `lint-staged` and works for me out of the box (I dont have prettier installed globally as well)

Comment (amit-mb) [reply package.json]: However, `jsonlint` doesn't work out of box although that too is a `devDependency` for `lint-staged`. it asks me to do `npm install jsonlint`. Not sure whats going on

Comment (revathskumar) [reply package.json]: Looks like [prettier](https://github.com/okonet/lint-staged/blob/d52e074f66ae3c604413fca48f0ba6b999c2fcfb/package.json#L64) is available as dependency

File: package.json [hunk 5]
```diff
@@ -66,7 +72,18 @@
...
+    "test:watch": "jest --watch",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "linters":{
+       "*.{js}": [
+        "prettier --config .prettierrc --write",
+        "git add"
+      ],
+       "*.{ts,tsx}": ["prettier --config .prettierrc --parser typescript --write", "git add", "jest"]
```
Comment (revathskumar) [inline package.json]: in `.prettierrc` we already specifying parser as typescript. So I think we don't need to specify this here

File: package.json [hunk 6]
```diff
@@ -66,7 +72,18 @@
     "build": "webpack -p --config webpack.config.prod.js",
     "watch": "webpack --config webpack.config.dev.js --watch --progress --colors",
     "test": "jest",
-    "test:watch": "jest --watch"
+    "test:watch": "jest --watch",
+    "precommit": "lint-staged"
+  },
+  "lint-staged": {
+    "linters":{
+       "*.{js}": [
```
Comment (revathskumar) [inline package.json]: Please club this along with `ts` & `tsx`. One rule will be enough

**Pull request merged**